### PR TITLE
Oxidation device file

### PR DESCRIPTION
**Pull request by TimeEngineer**
# What has changed ?

- [ ] Does this PR affect the User ?

### CODE REVIEW

**Review by Volham22 — CHANGES_REQUESTED** [review]

File: src/data/local_device_file.rs [hunk 1]
```diff
@@ -0,0 +1,153 @@
+use pyo3::{
```
[the remark this answers is not in the document]
Comment (TimeEngineer) [reply src/data/local_device_file.rs:1]: True

File: tests/common/smartcard.py [hunk 2]
```diff
@@ -209,19 +215,23 @@ def save_device_with_smartcard(
...
-            def _encrypt_dump(cleartext: bytes) -> Tuple[DeviceFileType, bytes, dict]:
-                extra_args = {
-                    "encrypted_key": b"123",
-                    "certificate_id": certificate_id,
-                    "certificate_sha1": certificate_sha1,
-                }
-                return DeviceFileType.SMARTCARD, cleartext, extra_args
+            def _encrypt_dump(
+                cleartext: bytes,
+            ) -> Tuple[DeviceFileType, bytes, bytes | None, bytes | None, str | None, bytes | None]:
```
[the remark this answers is not in the document]
Comment (TimeEngineer) [reply tests/common/smartcard.py]: Done

File: src/data/mod.rs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 mod certif;
```
[the remark this answers is not in the document]
Comment (TimeEngineer) [reply src/data/mod.rs:3]: Done

File: src/data/local_device_file.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (TimeEngineer) [reply src/data/local_device_file.rs:3]: Done

**Review by Volham22 — APPROVED** [review]

**Review by FirelightFlagboy — APPROVED** [review]

**Review by FirelightFlagboy — APPROVED** [review]

**Pull request merged**